diff --git a/src/display/wlr_window_manager.cc b/src/display/wlr_window_manager.cc
index 0000000..0000000 100644
--- a/src/display/wlr_window_manager.cc
+++ b/src/display/wlr_window_manager.cc
@@ -247,8 +247,8 @@ void WlrWindowManager::state(
   windows_[window]->minimized = false;
   windows_[window]->maximized = false;
   windows_[window]->fullscreen = false;
-  void *state_entry;
-  wl_array_for_each(state_entry, state) {
-    auto* entry = static_cast<uint32_t*>(state_entry);
-    if (*entry == ZWLR_FOREIGN_TOPLEVEL_HANDLE_V1_STATE_MAXIMIZED) {
+  uint32_t *state_entry;
+  wl_array_for_each(state_entry, state) {
+    uint32_t entry = *state_entry;
+    if (entry == ZWLR_FOREIGN_TOPLEVEL_HANDLE_V1_STATE_MAXIMIZED) {
       if (!windows_[window]->minimized) {
         windows_[window]->maximized = true;
       }
@@ -256,7 +256,7 @@ void WlrWindowManager::state(
     if (*entry == ZWLR_FOREIGN_TOPLEVEL_HANDLE_V1_STATE_MINIMIZED) {
       windows_[window]->minimized = true;
       windows_[window]->maximized = false;
       windows_[window]->fullscreen = false;
