--- src/display/wlr_window_manager.cc.orig
+++ src/display/wlr_window_manager.cc
@@ -254,8 +254,8 @@ void WlrWindowManager::state(
   windows_[window]->minimized = false;
   windows_[window]->maximized = false;
   windows_[window]->fullscreen = false;
-  void *state_entry;
-  wl_array_for_each(state_entry, state) {
-    auto* entry = static_cast<uint32_t*>(state_entry);
-    if (*entry == ZWLR_FOREIGN_TOPLEVEL_HANDLE_V1_STATE_MAXIMIZED) {
+  uint32_t *state_entry;
+  wl_array_for_each(state_entry, state) {
+    uint32_t entry = *state_entry;
+    if (entry == ZWLR_FOREIGN_TOPLEVEL_HANDLE_V1_STATE_MAXIMIZED) {
       if (!windows_[window]->minimized) {
         windows_[window]->maximized = true;
       }
@@ -263,7 +263,7 @@ void WlrWindowManager::state(
     if (*entry == ZWLR_FOREIGN_TOPLEVEL_HANDLE_V1_STATE_MINIMIZED) {
       windows_[window]->minimized = true;
